src/builder.rs

```diff
@@ -165,26 +166,21 @@ impl SessionBuilder {
         // if the call _didn't_ error, then the backgrounded ssh client will still hold onto those
         // handles, and it's still running, so those reads will hang indefinitely.
         let mut child = init.spawn().map_err(Error::Connect)?;
-        let status = child.wait().map_err(Error::Connect)?;
-
-        if let Some(255) = status.code() {
-            // this is the ssh command's way of telling us that the connection failed
-            let mut stderr = String::new();
-            child
-                .stderr
-                .as_mut()
-                .unwrap()
-                .read_to_string(&mut stderr)
-                .unwrap();
-
-            return Err(Error::interpret_ssh_error(&stderr));
+        let stdout = child.stdout.take().unwrap();
+        let mut stderr = child.stderr.take().unwrap();
+        let status = child.await.map_err(Error::Connect)?;
+
+        if !status.success() {
```

Comment:
This is different than the old check of `if let Some(255)` right? Are the semantics with `tokio::process` different?

Reply:
Well, normally `ssh` uses 255 for if ssh itself failed, and otherwise it just proxies the error of the remote command. When starting a master though, there is no remote command, so any error is due to ssh. While they _should_ all still be 255 (I think), it didn't seem worthwhile to check for that specifically here.